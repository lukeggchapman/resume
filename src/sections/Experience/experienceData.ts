import mdToHast from 'utils/mdToHast'

import assignarData from 'content/experience/2019-assignar.md'
import assignarLogo from 'content/experience/assignar.jpg'
import learnosityData from 'content/experience/2015-learnosity.md'
import learnosityLogo from 'content/experience/learnosity.png'
import componoData from 'content/experience/2020-compono.md'
import componoLogo from 'content/experience/compono.png'

const experienceData = [
  {
    frontmatter: componoData.data,
    logo: componoLogo,
    htmlAst: mdToHast(componoData.content),
  },
  {
    frontmatter: assignarData.data,
    logo: assignarLogo,
    htmlAst: mdToHast(assignarData.content),
  },
  {
    frontmatter: learnosityData.data,
    logo: learnosityLogo,
    htmlAst: mdToHast(learnosityData.content),
  },
]

export default experienceData
