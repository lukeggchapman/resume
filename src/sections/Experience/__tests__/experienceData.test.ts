import experienceData from '../experienceData'

jest.mock('utils/mdToHast', () => (md: string) => `mdToHast(${md})`)
jest.mock(
  'content/experience/2019-assignar.md',
  () => ({ data: 'assignarFrontmatterMock', content: 'assignarContentMock' }),
  {
    virtual: true,
  }
)
jest.mock(
  'content/experience/2018-psyfari.md',
  () => ({ data: 'psyfariFrontmatterMock', content: 'psyfariContentMock' }),
  {
    virtual: true,
  }
)
jest.mock(
  'content/experience/2015-learnosity.md',
  () => ({
    data: 'learnosityFrontmatterMock',
    content: 'learnosityContentMock',
  }),
  {
    virtual: true,
  }
)

describe('experienceData', () => {
  it('should have the correct data', () => {
    expect(experienceData).toMatchSnapshot()
  })
})
