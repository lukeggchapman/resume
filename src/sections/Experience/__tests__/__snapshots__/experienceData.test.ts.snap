// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`experienceData should have the correct data 1`] = `
[
  {
    "frontmatter": {
      "frontmatterFor": "indebtedContentMock",
    },
    "htmlAst": "htmlAst(indebtedContentMock)",
    "logo": "test-file-stub",
  },
  {
    "frontmatter": {
      "frontmatterFor": "pathzeroContentMock",
    },
    "htmlAst": "htmlAst(pathzeroContentMock)",
    "logo": "test-file-stub",
  },
  {
    "frontmatter": {
      "frontmatterFor": "componoContentMock",
    },
    "htmlAst": "htmlAst(componoContentMock)",
    "logo": "test-file-stub",
    "style": {
      "marginBottom": "100px",
    },
  },
  {
    "frontmatter": {
      "frontmatterFor": "assignarContentMock",
    },
    "htmlAst": "htmlAst(assignarContentMock)",
    "logo": "test-file-stub",
  },
  {
    "frontmatter": {
      "frontmatterFor": "learnosityContentMock",
    },
    "htmlAst": "htmlAst(learnosityContentMock)",
    "logo": "test-file-stub",
  },
]
`;
